Accessible mega nav

frrrances opened the pull request:
This PR makes the main mega nav for the libraries more accessible by adding structure, aria labels and controls, and some keyboard functionality for opening and closing the menus.

frrrances commented:
This branch is up on dev right now.

matt-bernhardt commented on footer.php:
```diff
@@ -130,6 +132,49 @@
+                        break;
+
+            // user presses the "esc" key
+            case 27:    hideMenu(e,);
```
This line is throwing a syntax error in Firefox and IE on Windows. `expected expression, got ')'`

frrrances replied:
Nice catch! Fixed it... let me know if that helps things.

frrrances commented:
@matt-bernhardt Thanks for the good review. I addressed the extra comma, the subnav showing on mobile menu hover issue, and fixed the tests. Let me know if there's anything else I should address.

frrrances commented:
@matt-bernhardt  I moved the js into the grunt files. Let me know if this looks good.
